x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/users/managed_details/index.ts

```diff
@@ -7,33 +7,65 @@
 
 import type { IEsSearchResponse } from '@kbn/data-plugin/common';
 
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { getOr } from 'lodash/fp';
 import { inspectStringifyObject } from '../../../../../utils/build_query';
 import type { SecuritySolutionFactory } from '../../types';
 import { buildManagedUserDetailsQuery } from './query.managed_user_details.dsl';
 
 import type { UsersQueries } from '../../../../../../common/search_strategy/security_solution/users';
 import type {
-  AzureManagedUser,
+  ManagedUserHits,
+  ManagedUserHit,
   ManagedUserDetailsStrategyResponse,
+  ManagedUserFields,
 } from '../../../../../../common/search_strategy/security_solution/users/managed_details';
+import { ManagedUserDatasetKey } from '../../../../../../common/search_strategy/security_solution/users/managed_details';
+
+interface ManagedUserBucket {
+  key: ManagedUserDatasetKey;
+  latest_hit: SearchResponse<ManagedUserFields | ManagedUserFields>;
+}
 
 export const managedUserDetails: SecuritySolutionFactory<UsersQueries.managedDetails> = {
   buildDsl: (options) => buildManagedUserDetailsQuery(options),
   parse: async (
     options,
-    response: IEsSearchResponse<AzureManagedUser>
+    response: IEsSearchResponse<ManagedUserFields>
   ): Promise<ManagedUserDetailsStrategyResponse> => {
     const inspect = {
       dsl: [inspectStringifyObject(buildManagedUserDetailsQuery(options))],
     };
 
-    const hits = response.rawResponse.hits.hits;
-    const userDetails = hits.length > 0 ? hits[0]._source : undefined;
+    const buckets: ManagedUserBucket[] = getOr(
+      [],
+      'aggregations.datasets.buckets',
+      response.rawResponse
+    );
+
+    const managedUsers: ManagedUserHits = buckets.reduce(
+      (acc: ManagedUserHits, bucket: ManagedUserBucket) => {
+        acc[bucket.key] = bucket.latest_hit.hits.hits[0] as unknown as ManagedUserHit;
```

Comment:
Isn't there any way to avoid this ugly casting? Both `ManagedUserBucket` and `ManagedUserHit` are defined by us.

Reply:
I am fixing this issues here: https://github.com/elastic/kibana/pull/172650